drivers/linstorvolumemanager.py

```diff
@@ -196,18 +200,57 @@ def _get_controller_uri():
         # Not found, maybe we are trying to create the SR...
         pass
 
-def get_controller_uri():
-    retries = 0
-    while True:
+
+def get_cached_controller_uri():
+    try:
+        with open(CONTROLLER_CACHE_PATH, "r") as f:
+            return f.read().strip()
+    except Exception as e:
+        util.SMlog('Unable to read controller uri cache file at {} : {}'.format(
+            CONTROLLER_CACHE_PATH,
+            e
+        ))
+    return None
+
+def delete_controller_uri_cache():
+    try:
+        os.remove(CONTROLLER_CACHE_PATH)
+    except Exception as e:
+        util.SMlog('Unable to write controller uri cache file at {} : {}'.format(
+            CONTROLLER_CACHE_PATH,
+            e
+        ))
+
+def write_controller_uri_cache(uri):
+    try:
+        if not os.path.exists(CONTROLLER_CACHE_DIRECTORY):
+            os.makedirs(CONTROLLER_CACHE_DIRECTORY)
+            os.chmod(CONTROLLER_CACHE_DIRECTORY, 0o700)
+        if not os.path.isdir(CONTROLLER_CACHE_DIRECTORY):
+            raise NotADirectoryError
```

Comment:
I'm not sure that's important here. You will have an error raised by the next open call if we don't manipulate a file.